esphome/components/wts01/wts01.cpp

```diff
@@ -0,0 +1,99 @@
+#include "wts01.h"
+#include "esphome/core/log.h"
+#include <cmath>
+
+namespace esphome {
+namespace wts01 {
+
+static const char *const TAG = "wts01";
+static const uint8_t PACKET_SIZE = 9;
+static const uint8_t HEADER_1 = 0x55;
+static const uint8_t HEADER_2 = 0x01;
+static const uint8_t HEADER_3 = 0x01;
+static const uint8_t HEADER_4 = 0x04;
+
+void WTS01Sensor::loop() {
+  // Process one character at a time received from the sensor
+  if (available()) {
+    uint8_t c;
+    if (read_byte(&c)) {
```

Comment:
> All uses of class members and member functions should be prefixed with this-> to distinguish them from global functions/variables.

https://developers.esphome.io/contributing/code/#c-code-style
```suggestion
  if (this->available()) {
    uint8_t c;
    if (this->read_byte(&c)) {
```